### ISSUE TITLE
Install script should not attempt to write a launcher script if "$HOME/.meteor" is in my PATH

### ISSUE THREAD

**Issue by parisk** [meteor/meteor#4345]
In case I have already `$HOME/.meteor` in my `PATH`, the installer should not attempt write a launcher script in `/usr/local/bin`.

The main reasons behind this are:
- this is completely useless; the `meteor` executable will be already accessible in my shell, since it resides in `$HOME/.meteor` and since I already have `$HOME/.meteor` in my path. No added value here.
- this makes it harder to automate meteor installations in systems, since it will ask for `sudo` password to gain `root` access, that is not essentially needed and some users might not have it.

**Bottom line**: I believe that `root` access during installation should be asked only when it is essentially needed.

I would be more than happy to submit a patch for this if you are okay with such a case.

**Comment by stubailo** [comment]
The install script is not very complex - if you would like a most customizable, automatable installation, you can simply reuse part of the existing script and download the installation tarball from the same place.

Read the script at http://install.meteor.com/

**Comment by Slava** [comment]
I think this PR is related: https://github.com/meteor/meteor/pull/4330

**Comment by glasser** [comment]
This does seem reasonable.  The install script is already pretty complicated, but sure, in the relatively rare case where that directory is already in PATH we could print a relevant message and stop.  Pull requests encouraged (or, well, post a diff --- the install.meteor.com script is not in a public repository).  Don't forget that this script uses sh, not bash, so don't use any bash-specific features (eg, test with dash).

**Comment by TomFreudenberg** [comment]
IMHO you should add an option to the installer to give that decision to the one who is calling the installer

like `--install-only-local`

what will not run the `sudo` options

**Comment by asl97** [comment]
it can't be that complex.

```
133a134,140
> case "$PATH" in
> # .not fool prove but it work
> *"$HOME/.meteor"*) exit 0 ;;
> # install the launcher in the home bin if it is in $PATH instead of a system wide /usr/local/bin
> *"$HOME/bin"*       ) PREFIX="$HOME" ;;
> esac
> 
```

**Comment by parisk** [comment]
I would like to give this a try. Is the installer located in this repository? I was not able to find it.

**Comment by TomFreudenberg** [comment]
Hi @parisk 

no, the installer is NOT in the repository.

You just may _curl_ the file downto you system and can edit it. As @glasser wrote, you may then send in a diff or a patch file as attachment here.

```
curl https://install.meteor.com -o install.sh
```

then you get the `install.sh` file on your disk.

Please check the `sudo` elements in `install.sh` - They must be _healed_ if you want change that behavior. 

Cheers
Tom

**Comment by hwillson** [comment]
To help provide a more clear separation between feature requests and bugs, and to help clean up the feature request backlog, Meteor feature requests are now being managed under the https://github.com/meteor/meteor-feature-requests repository. 

This feature request will be closed here, but anyone interested in migrating this feature request to the new repository (to make sure it stays active), can click [here](https://github.com/meteor/meteor-feature-requests/issues/new?title=Install%20script%20should%20not%20attempt%20to%20write%20a%20launcher%20script%20if%20%22%24HOME%2F.meteor%22%20is%20in%20my%20PATH&body=Migrated%20from:%20meteor/meteor%234345) to start the feature request migration process. This manual migration process is intended to help identify which of the older feature requests are still considered to be of value to the community. Thanks!
